### PR TITLE
[4.0] Move sidebar modules into grid

### PR DESCRIPTION
**Pull request by ciar4n**
Pull Request for Issue # .

### Summary of Changes
Moves the sidebar modules into the main CSS grid by using `has-*` classes.

- Less nested divs
- Less hacky
- Potential scope for IE11 CSS grid support (using old grid spec). 

For a more detailed explanation.. https://joomla51.com/blog/item/joomla-css-grid-and-dynamic-content

### Testing Instructions
Apply patch (cant use patchtester) and compile CSS. Check frontend layout with modules enabled/disabled in the **sidebar-left** and **sidebar-right** modules positions

### CODE REVIEW

File: templates/cassiopeia/index.php [hunk 1]
```diff
@@ -50,6 +50,14 @@
 	$logo = '<img src="' . $this->baseurl . '/templates/' . $this->template . '/images/logo.svg' . '" class="logo d-inline-block" alt="' . $sitename . '">';
 }
 
+$hasClass = '';
+if ($this->countModules('sidebar-left')) {
```
Comment (Quy) [inline templates/cassiopeia/index.php]:
```suggestion

if ($this->countModules('sidebar-left'))
{
```

File: templates/cassiopeia/index.php [hunk 2]
```diff
@@ -50,6 +50,14 @@
...
+ $hasClass .= ' has-sidebar-left';
+}
+if ($this->countModules('sidebar-right')) {
```
Comment (Quy) [inline templates/cassiopeia/index.php]:
```suggestion

if ($this->countModules('sidebar-right'))
{
```

File: templates/cassiopeia/index.php [hunk 3]
```diff
@@ -50,6 +50,14 @@
 	$logo = '<img src="' . $this->baseurl . '/templates/' . $this->template . '/images/logo.svg' . '" class="logo d-inline-block" alt="' . $sitename . '">';
 }
 
+$hasClass = '';
+if ($this->countModules('sidebar-left')) {
+ $hasClass .= ' has-sidebar-left';
```
Comment (Quy) [inline templates/cassiopeia/index.php]:
```suggestion
	$hasClass .= ' has-sidebar-left';
```

File: templates/cassiopeia/index.php [hunk 4]
```diff
@@ -50,6 +50,14 @@
...
+}
+if ($this->countModules('sidebar-right')) {
+ $hasClass .= ' has-sidebar-right';
```
Comment (Quy) [inline templates/cassiopeia/index.php]:
```suggestion
	$hasClass .= ' has-sidebar-right';
```

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on 3e4f52ef0b050abfa798d6f5b6c38fccea5e8f03<br /><br />Code is cleaner after applying the patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23661">issues.joomla.org/tracker/joomla-cms/23661</a>.</sub>

**Comment by ciar4n** [comment]
Appreciate if I could get a second test on this. Without it, the grid is just a single column. Defeats the purpose of using CSS grid. Might as well just use flex.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 1088ffb4ab919128c1193c568d0fc3eb14a9c8db<br /><br />Works for me.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23661">issues.joomla.org/tracker/joomla-cms/23661</a>.</sub>

**Comment by ciar4n** [comment]
@roland-d @infograf768 Thank you. RTC?

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23661">issues.joomla.org/tracker/joomla-cms/23661</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**